Sources/Misc/SystemInfo.swift

```diff
@@ -126,12 +126,27 @@ class SystemInfo {
             : BundleSandboxEnvironmentDetector(bundle: bundle)
     }
 
+    /// Asynchronous API if caller can't ensure that it's invoked in the `@MainActor`
+    /// - Seealso: `isApplicationBackgrounded`
     func isApplicationBackgrounded(completion: @escaping (Bool) -> Void) {
-        self.operationDispatcher.dispatchOnMainThread {
+        self.operationDispatcher.dispatchOnMainActor {
             completion(self.isApplicationBackgrounded)
         }
     }
 
+    /// Synchronous API for callers in `@MainActor`.
+    /// - Seealso: `isApplicationBackgrounded(completion:)`
+    @MainActor
```

Comment:
this seems like a good idea, it basically guarantees that we won't be able to call this on a thread other than main, right? Which would crash if we did, so now it fails in compilation instead

Reply:
Exactly.
Well it wouldn't have crashed I don't think, but `UIApplication` isn't thread safe so it wouldn't have been safe.

Reply:
sorry you're right, not a crash, but this [is what I was thinking about](https://github.com/RevenueCat/purchases-ios/issues/300). I remembered there being issues with it

Reply:
Oooh makes sense.